virtualizarr/writers/icechunk.py

```diff
@@ -329,13 +329,17 @@ def write_manifest_virtual_refs(
         op_flags=[["readonly"]] * 3,  # type: ignore
     )
 
+    # Icechunk checksums currently store with second precision, so we need to make sure
+    # the checksum_date is at least one second in the future
+    timestamp_checksum = datetime.now(timezone.utc) + timedelta(seconds=1)
+
```

Comment:
@mpiannucci I feel like this +1 second business should be in the test, rather than in the business logic?

Reply:
Yes I agree

Reply:
We actually came to the opposite conclusion on a call just now. The long comments in the code hopefully explain our reasoning.